server/src/main/java/org/apache/iotdb/db/query/workloadmanager/WorkloadManager.java

```diff
@@ -0,0 +1,155 @@
+  private static final WorkloadManager manager = new WorkloadManager();
+  List<QueryRecord> records = Collections.synchronizedList(new ArrayList<>());
+  private static final Logger logger = LoggerFactory.getLogger(WorkloadManager.class);
+  private int aggregationCount = 0;
+  private int groupByCount = 0;
+  private static final int RECORDS_SIZE = 100;
+  private static final int RECORD_FILE_MAX_SIZE = 1024 * 1024 * 1024;
+  private File curWritingFile = null;
+
+  public WorkloadManager() {
+  }
+
+  public static WorkloadManager getInstance() {
+    return manager;
+  }
+
+  public void addAggregationRecord(String device, List<String> sensorList, List<String> opList) {
+    // add aggregation record
+    String[] sensors = new String[sensorList.size()];
+    String[] ops = new String[opList.size()];
+
+    for (int i = 0; i < sensorList.size(); ++i) {
+      sensors[i] = sensorList.get(i);
+    }
+    for (int i = 0; i < opList.size(); ++i) {
+      ops[i] = opList.get(i);
+    }
+    QueryRecord record = new AggregationQueryRecord(device, sensors, ops);
+        /*if (!historyMap.containsKey(record)) {
+            historyMap.put(record, new AggregationHistory());
+        }
+        try {
+            historyMap.get(record).addRecord();
+        } catch (MismatchRecordTypeException e) {
+            e.printStackTrace();
+        }*/
+    records.add(record);
+  }
+
+  public void addGroupByQueryRecord(String device, List<String> sensorList, List<String> opList,
+                               long startTime, long endTime, long interval, long slidingStep) {
+    // add group by record
+    String[] sensors = new String[sensorList.size()];
+    String[] ops = new String[opList.size()];
+
+    for (int i = 0; i < sensorList.size(); ++i) {
+      sensors[i] = sensorList.get(i);
+    }
+    for (int i = 0; i < opList.size(); ++i) {
+      ops[i] = opList.get(i);
+    }
+
+    QueryRecord record = new GroupByQueryRecord(device, sensors, ops, startTime, endTime, interval, slidingStep);
+        /*if (!historyMap.containsKey(record)) {
+            historyMap.put(record, new GroupByHistory());
+        }
+        try {
+            historyMap.get(record).addRecord(startTime, endTime, interval, slidingStep);
+        } catch (MismatchRecordTypeException e) {
+            e.printStackTrace();
+        }*/
+    records.add(record);
+  }
+
+  public void addRecord(QueryRecord record) {
+    records.add(record);
+    if (records.size() >= RECORDS_SIZE) {
+      flushRecords();
+    }
+    switch (record.getRecordType()) {
+      case AGGREGATION: {
+        aggregationCount++;
+        break;
+      }
+      case GROUP_BY: {
+        groupByCount++;
+        break;
+      }
+    }
+    // logger.info("Record: " + record.getSqlWithTimestamp());
+    logger.info("Adding record type of " + record.getRecordType());
+    logger.info("Aggregation count: " + aggregationCount + ", group by count: " + groupByCount);
+  }
+
+  private void flushRecords() {
```

Comment:
There is no need to choose what file you write into. You can define your own logger, and the LOGGER will do everything for you, like automatically switch to a new log file while the current file reach the thredshold. You can refer to the `SLOW_SQL_LOGGER` field in the `TSServiceImpl` class and its logger action is defined in `server/src/assembly/resources/conf/logback.xml`.